### PR TITLE
Работа с базой данных выполняется в фоновом потоке.

### PR DESCRIPTION
**Pull request by Kalyanova**
Работа с базой данных выполняется в фоновом потоке. Чтобы добиться этого и правильно обновлять пользовательский интерфейс, приложение использует ViewModel и LiveData.

### CODE REVIEW

File: app/src/main/java/by/paranoidandroid/notesapp/activities/MainActivity.java [hunk 1]
```diff
@@ -28,27 +30,29 @@ protected void onCreate(Bundle savedInstanceState) {
         recyclerView.setLayoutManager(new LinearLayoutManager(this));
         recyclerView.addItemDecoration(new DividerItemDecoration(this, DP_MARGIN));
 
-        adapter = new NotesAdapter(this);
+        adapter = new NotesAdapter(this, this);
```
Comment (vally) [inline app/src/main/java/by/paranoidandroid/notesapp/activities/MainActivity.java:33]: Not readable, I suggest to rename them

File: app/src/main/java/by/paranoidandroid/notesapp/database/AppDatabase.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public static AppDatabase getAppDatabase(Context context) {
                             // Allow queries on the main thread.
                             // Don't do this on a real app!
                             // Only for testing purposes.
-                            .allowMainThreadQueries()
+                            //.allowMainThreadQueries()
```
Comment (vally) [inline app/src/main/java/by/paranoidandroid/notesapp/database/AppDatabase.java:29]: remove it

File: app/src/main/java/by/paranoidandroid/notesapp/adapters/NotesAdapter.java [hunk 3]
```diff
@@ -11,18 +11,20 @@
...
+import by.paranoidandroid.notesapp.utils.RemoveItemListener;
 
 public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {
     private List<Note> notes;
     private LayoutInflater inflater;
     private Context context;
+    private RemoveItemListener removeItemListener;
 
-    public NotesAdapter(Context ctx) {
+    public NotesAdapter(Context ctx, RemoveItemListener listener) {
```
Comment (vally) [inline app/src/main/java/by/paranoidandroid/notesapp/adapters/NotesAdapter.java:23]: Think, how can do without the listener

File: app/src/main/java/by/paranoidandroid/notesapp/viewmodel/NoteViewModel.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        public <T extends ViewModel> T create(Class<T> modelClass) {
+            //noinspection unchecked
+            return (T) new NoteViewModel(application);
+        }
+    }
+
+    /**
+     * AsyncTask that executes asynchronous operations (queries to database) in background.
+     */
+    private static class DBAsyncTask extends AsyncTask<Note, Void, List<Note>> {
```
Comment (vally) [inline app/src/main/java/by/paranoidandroid/notesapp/viewmodel/NoteViewModel.java:69]: Use AsyncTaskLoader is better

File: app/src/main/java/by/paranoidandroid/notesapp/activities/MainActivity.java [hunk 5]
```diff
@@ -28,27 +30,29 @@ protected void onCreate(Bundle savedInstanceState) {
...
         recyclerView.setAdapter(adapter);
-        adapter.updateList(getNoteList());
 
         etTitle =  findViewById(R.id.et_title);
         etBody =  findViewById(R.id.et_body);
 
         Button button = findViewById(R.id.bt_add_note);
         button.setOnClickListener(view -> onAddNoteButtonClick());
+
+        NoteViewModel.Factory viewModelFactory = new NoteViewModel.Factory(getApplication());
```
Comment (vally) [inline app/src/main/java/by/paranoidandroid/notesapp/activities/MainActivity.java:42]: Why did you create a factory?

**Review by vally — COMMENTED** [review]
Look at my comments please

File: app/src/main/java/by/paranoidandroid/notesapp/database/AppDatabase.java (hunk 2 above)
Comment (Kalyanova) [reply app/src/main/java/by/paranoidandroid/notesapp/database/AppDatabase.java:29]: Fixed in https://github.com/Kalyanova/NotesApp/pull/3.

File: app/src/main/java/by/paranoidandroid/notesapp/activities/MainActivity.java (hunk 5 above)
Comment (Kalyanova) [reply app/src/main/java/by/paranoidandroid/notesapp/activities/MainActivity.java:42]: Yes, this factory is completely useless, removed in https://github.com/Kalyanova/NotesApp/pull/3.

File: app/src/main/java/by/paranoidandroid/notesapp/activities/MainActivity.java (hunk 1 above)
Comment (Kalyanova) [reply app/src/main/java/by/paranoidandroid/notesapp/activities/MainActivity.java:33]: Not readable is namely "NotesAdapter" name, should this adapter be renamed? Should it be more verbose?

File: app/src/main/java/by/paranoidandroid/notesapp/adapters/NotesAdapter.java (hunk 3 above)
Comment (Kalyanova) [reply app/src/main/java/by/paranoidandroid/notesapp/adapters/NotesAdapter.java:23]: Unfortunately, I seem to have no idea how to do this in other way... Could you please give me a tip?

File: app/src/main/java/by/paranoidandroid/notesapp/viewmodel/NoteViewModel.java (hunk 4 above)
Comment (Kalyanova) [reply app/src/main/java/by/paranoidandroid/notesapp/viewmodel/NoteViewModel.java:69]: And I don't fully understand why AsyncTaskLoader is better. Could you explain please?

**Pull request merged**